### PR TITLE
Adding the reverberation script to AMI

### PR DESCRIPTION
**Pull request by tomkocse**

### CODE REVIEW

**Comment by danpovey** [comment]
This is very nice!  I'll wait for Vijay to check and merge it.

File: egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+fi
+
+if [ $stage -le 13 ]; then
+  # Get the alignments as lattices (gives the chain training more freedom).
+  # use the same num-jobs as the alignments
+  steps/align_fmllr_lats.sh --nj 100 --cmd "$train_cmd" ${lores_train_data_dir} \
+    data/lang $gmm_dir $original_lat_dir
+  rm $original_lat_dir/fsts.*.gz # save space
+
+if [ "$add_reverberated_data" == "true" ]; then
```
Comment (vijayaditya) [inline egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh]: correct indentation.

File: egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+set -e -o pipefail
+
+# First the options that are passed through to run_ivector_common.sh
+# (some of which are also used in this script directly).
+stage=1
+mic=ihm
+nj=30
+min_seg_len=1.55
+use_ihm_ali=true
```
Comment (vijayaditya) [inline egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh]: As you are already in the multi-condition directory you can make add_reverberated_Data compulsary. Also if you are using add_reverberated_data option use_ihm_ali would be true*. So you could eliminate both these options.

*As you are already assuming availability of IHM data it does not make sense to not use it for alignments.

File: egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh [hunk 3]
```diff
@@ -0,0 +1,280 @@
+#!/bin/bash
+
+# This is a chain-training script with TDNN neural networks.
+# This script inherits from local/chain/run_tdnn.sh, but adding
```
Comment (vijayaditya) [inline egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh]: It is based on local/chain/run_tdnn.sh

File: egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    --xent-separate-forward-affine true \
+    --include-log-softmax false \
+    --final-layer-normalize-target 1.0 \
+   $dir/configs || exit 1;
+fi
+
+if [ $stage -le 16 ]; then
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $dir/egs/storage ]; then
+    utils/create_split_dir.pl \
+     /export/b0{5,6,7,8}/$USER/kaldi-data/egs/ami-$(date +'%m_%d_%H_%M')/s5b/$dir/egs/storage $dir/egs/storage
```
Comment (vijayaditya) [inline egs/ami/s5b/local/chain/multi_condition/run_tdnn.sh]: ami-rvb

File: egs/ami/s5b/local/nnet3/multi_condition/run_ivector_common.sh [hunk 5]
```diff
@@ -0,0 +1,250 @@
+#!/bin/bash
+
+set -e -o pipefail
+
+
+# This script is called from local/nnet3/run_tdnn.sh and local/chain/run_tdnn.sh (and may eventually
```
Comment (vijayaditya) [inline egs/ami/s5b/local/nnet3/multi_condition/run_ivector_common.sh]: correct script paths.

File: egs/ami/s5b/local/nnet3/multi_condition/run_ivector_common.sh [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+                  # it should contain alignments for 'train_set'.
+
+
+num_threads_ubm=32
+rvb_affix=_rvb
+nnet3_affix=_cleaned     # affix for exp/$mic/nnet3 directory to put iVector stuff in, so it
+                         # becomes exp/$mic/nnet3_cleaned or whatever.
+num_data_reps=1
+
+. cmd.sh
```
Comment (vijayaditya) [inline egs/ami/s5b/local/nnet3/multi_condition/run_ivector_common.sh]: . ./cmd.sh

File: egs/ami/s5b/local/nnet3/multi_condition/run_ivector_common.sh [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+      ${datadir}_rvb${num_data_reps}_hires $min_seg_len ${datadir}_rvb${num_data_reps}_hires_comb
+
+    # just copy over the CMVN to avoid having to recompute it.
+    cp ${datadir}_rvb${num_data_reps}_hires/cmvn.scp ${datadir}_rvb${num_data_reps}_hires_comb/
+    utils/fix_data_dir.sh ${datadir}_rvb${num_data_reps}_hires_comb/
+  fi
+
+  utils/combine_data.sh data/${mic}/${train_set}_sp_rvb_hires data/${mic}/${train_set}_sp_hires ${datadir}_rvb${num_data_reps}_hires
+  utils/combine_data.sh data/${mic}/${train_set}_sp_rvb_hires_comb data/${mic}/${train_set}_sp_hires_comb ${datadir}_rvb${num_data_reps}_hires_comb
+exit
```
Comment (vijayaditya) [inline egs/ami/s5b/local/nnet3/multi_condition/run_ivector_common.sh]: remove the exit.

**Comment by vijayaditya** [comment]
@tomkocse Could you also add results for mdm8 ?

**Pull request merged**